src/status_im/chat/models/input.cljs

```diff
@@ -219,22 +228,21 @@
 
 (rf/defn send-edited-message
   [{:keys [db] :as cofx} text {:keys [message-id quoted-message chat-id]}]
-  (let [pinned-message (get-in db [:pin-messages chat-id message-id])]
-    (rf/merge
-     cofx
-     {:json-rpc/call [{:method      "wakuext_editMessage"
-                       :params      [{:id           message-id
-                                      :text         text
-                                      :content-type (if (message-content/emoji-only-content?
-                                                         {:text text :response-to quoted-message})
-                                                      constants/content-type-emoji
-                                                      constants/content-type-text)}]
-                       :js-response true
-                       :on-error    #(log/error "failed to edit message " %)
-                       :on-success  (fn [result]
-                                      (re-frame/dispatch [:sanitize-messages-and-process-response
-                                                          result]))}]}
-     (cancel-message-edit))))
+  (rf/merge
```

Comment:
**[Nothing to be done in this PR]**

It's quite unfortunate that we keep this `rf/defn` macro in the codebase. As we can see here, the linter is unable to lint handler functions. We can type garbage inside event handlers and nothing is verified. This is exacerbated by the fact that status-mobile event handlers are not unit tested...

From now on, every time I see a situation where this macro & pattern causes pain I'm gonna link to the discussion https://github.com/status-im/status-mobile/discussions/15099 (without traction unfortunately) created by @ulisesmac some time ago. I regret not having done this up until this point :shrug: Over time, I hope we'll have more evidence about the downsides and we can get together and see what can be done to improve the macro or ditch it away.

Reply:
> the linter is unable to lint handler functions

Oh I see. I saw `pinned-message` which is not used in the function, I wondered why the linter did not warn against it.